### PR TITLE
add gmm

### PR DESCRIPTION
**Pull request by yang71**
add gmm

### CODE REVIEW

File: examples/gmm/gmm_trainer.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    # 0.001、0.01、0.1（0.1绝对不可，目前试下来0.01比较好）
+    parser.add_argument("--lr", type=float, default=0.01, help="learning rate")
+    # cora 0.5-0.9（目前试下来0.9表现最好） pubmet 0.4-0.8
+    parser.add_argument("--drop_rate", type=float, default=0.9, help="drop_rate")
+    parser.add_argument("--n_epoch", type=int, default=200, help="number of epoch")
+    parser.add_argument("--hidden_dim", type=int, default=32, help="dimention of hidden layers")  # 16\32\64 (32比较好)
+    parser.add_argument("--num_layers", type=int, default=2, help="number of layers")  # 1、2 （2的表现更好）
+    parser.add_argument("--pseudo_dim", type=int, default=2, help="Pseudo coordinate dimensions in GMMConv")
+    parser.add_argument("--n_kernels", type=int, default=3, help="Number of kernels in GMMConv layer")
+    parser.add_argument("--l2_coef", type=float, default=5e-3, help="l2 loss coeficient")  # 1、2、5/-5、-4、-3(-3、-4好像都还可以)
```
Comment (dddg617) [inline examples/gmm/gmm_trainer.py:146]: Remember to replace Chinese with English.

File: gammagl/layers/conv/gmm_conv.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+if __name__ == "__main__":
+    edge_index = tlx.convert_to_tensor([[0, 1, 2, 3, 6, 2, 5],
+                                        [1, 2, 3, 4, 2, 0, 3]], dtype=tlx.int64)
+    # u, v = edge_index[0, :], edge_index[1, :]
+    edge_index, _ = add_self_loops(edge_index)
+    x = tlx.ones((7, 10))  # x就是feat
+    conv = GMMConv(10, 2, 2, 3, 'mean')
+    pseudo = tlx.ones((14, 2))
+    res = conv(x, edge_index, pseudo)
+    print(res)
```
Comment (dddg617) [inline gammagl/layers/conv/gmm_conv.py:86]: I think we do not need the __main__, remove these lines.

File: gammagl/layers/conv/gmm_conv.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                                    out_features=n_kernels * out_channels,
+                                    W_init='xavier_uniform',
+                                    b_init=None)
+        self.pseudo_fc = tlx.layers.Linear(in_features=dim,
+                                    out_features=2,
+                                    W_init='xavier_uniform')
+        if self.add_bias:
+            initor = tlx.initializers.Zeros()
+            self.bias = self._get_weights("bias", shape=(1, self.out_channels), init=initor)
+    def forward(self, x, edge_index, pseudo):
```
Comment (dddg617) [inline gammagl/layers/conv/gmm_conv.py:45]: Why do we need the parameter *pseudo*? I do not find the parameters in PyG.

File: gammagl/layers/conv/gmm_conv.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        pseudo = tlx.reshape(pseudo, (E, 1, self.dim))  # 13264*2->13264*1*2
+        # self.mu = tlx.reshape(self.mu, (1, self.n_kernels, self.dim))  # 1*3*2
+        gaussian = -0.5 * ((pseudo - self.mu) ** 2)  # 13262*3*2
+        # self.sigma = tlx.reshape(self.sigma, (1, self.n_kernels, self.dim))  #  1*3*2
+        gaussian = gaussian * (self.sigma ** 2)  # 13264*3*2
+        # gaussian = tlx.exp(tlx.reduce_sum(gaussian, -1, keepdims=True))  # (E, K, 1)
+        gaussian = tlx.exp(tlx.reduce_sum(gaussian, -1))  # (E, K) 13264*3
+        weights = gaussian  # 将其作为边权重
+        out = self.propagate(src_feat, edge_index, edge_weight=weights, num_nodes=num_nodes, aggr=self.aggr)  # 2708*3*7?
+        out = tlx.reduce_sum(out, -2)  # 2708*7
```
Comment (dddg617) [inline gammagl/layers/conv/gmm_conv.py:71]: Remember to remove the comments that are of no use.

**Comment by dddg617** [comment]
[TODO]
1. Add readme.md in the example and put the hyperparameters there, you can refer to [this](https://github.com/BUPT-GAMMA/GammaGL/blob/main/examples/gcn/readme.md).
2. Add rst docstrings in the GMMConv, you can also refer to GCNConv.
3. Remove Chinese comments, useless comments, and files.

**Comment by yang71** [comment]
gmm modification.

File: gammagl/layers/conv/gmm_conv.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+        Returns
+        -------
+        torch.Tensor
+            The output feature of shape :math:`(N, D_{out})` where :math:`D_{out}`
+            is the output feature size.
+        """
+        src_feat, dst_feat = x, x
+        num_nodes = int(dst_feat.shape[0])
+        E = edge_index.shape[1]
```
Comment (dddg617) [inline gammagl/layers/conv/gmm_conv.py:96]: Please replace .shape with tlx.get_tensor_shape()

**Pull request closed without merging**